pil-analyzer/src/pil_analyzer.rs

```diff
@@ -46,6 +47,7 @@ fn analyze<T: FieldElement>(files: Vec<PILFile>) -> Analyzed<T> {
     let mut analyzer = PILAnalyzer::new();
     analyzer.process(files);
     analyzer.side_effect_check();
+    analyzer.match_exhaustiveness_check();
```

Comment:
I think the errors we get are better if we first do type checking, because we assume the types to be correct in the exhaustiveness check.